### PR TITLE
fix: harden unstack name resolution and rebase cleanup

### PR DESCRIPTION
**Pull request by mrmans0n**
## Summary

Follow-up to #309 addressing remaining Codex review feedback on `gg unstack`:

- **Config collision rejection**: `resolve_new_stack_name` now checks existing `.git/gg/config.json` stack entries before accepting a name (explicit or auto-generated), preventing silent overwrite of MR mappings or worktree metadata.
- **Bounded auto-name generation**: Introduced `AUTO_NAME_ATTEMPT_LIMIT = 100`. Generated candidates pass through `sanitize_stack_name`; invalid candidates are skipped. Trailing-dash stack names are handled by trimming before generating suffixes.
- **Rebase failure cleanup hardened**: Extracted `cleanup_failed_unstack_rebase` helper that aborts the rebase, checks out the original stack branch, deletes the temporary branch, and propagates actionable errors instead of silently ignoring failures. Handles detached-HEAD and no-rebase-in-progress edge cases.
- **Inherited base preserved** (confirmed as false positive): `migrate_config` correctly copies `base: None`, so stacks inheriting `defaults.base` remain inherited. Added a dedicated test to confirm.

## Test plan

- [x] 5 new unit tests covering config collision, auto-name generation bounds, trailing-dash handling, and base inheritance
- [x] \`cargo fmt --all\`
- [x] \`cargo clippy --all-targets --all-features -- -D warnings\`
- [x] \`cargo test -p gg-core --all-features\` (unstack tests pass)
- [x] \`cargo test -p gg-cli --test integration_tests unstack\` (integration tests pass)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/gg-core/src/commands/unstack.rs [hunk 1]
```diff
@@ -382,6 +388,40 @@ fn rebase_upper_stack(
...
+        .args(["rebase", "--abort"])
+        .output()?;
+    if !abort_output.status.success() && git::is_rebase_in_progress(repo) {
+        return Err(GgError::Other(format!(
+            "Rebase failed, and cleanup could not abort the rebase: {}",
+            String::from_utf8_lossy(&abort_output.stderr)
+        )));
+    }
+
+    git::checkout_branch(repo, original_branch).map_err(|e| {
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply crates/gg-core/src/commands/unstack.rs]: Good catch. Fixed in 8bf72cc — checkout errors are now captured but no longer block temp branch deletion. The checkout still happens (and its failure is reported), but the cleanup sequence completes regardless.

**Comment by mrmans0n** [comment]
Fixed the failed-rebase cleanup so checkout remains best-effort and detached/worktree conflict flows still surface the normal rebase conflict path.\n\n@codex review\n@claude review
